lib/rspec/shell/expectations/stubbed_env.rb

```diff
@@ -28,22 +28,78 @@ def cleanup
         end
 
         def stub_command(command)
+          wrap_command command
           StubbedCommand.new command, @dir
         end
 
         def execute(command, env_vars = {})
-          Open3.capture3(env_vars, "#{env} #{command}")
+          full_command=wrap_execute(<<-multiline_script
+            #{env} source #{command} 2> #{wrapped_error_path}
+          multiline_script
+          )
+
+          Open3.capture3(env_vars, full_command)
         end
 
         def execute_function(script, command, env_vars = {})
-          Open3.capture3(env_vars, "bash -c 'source #{script} && #{env} #{command}'")
+          full_command=wrap_execute(<<-multiline_script
+            source #{script} 2> #{wrapped_error_path}
+            #{env} #{command}
+          multiline_script
+          )
+
+          Open3.capture3(env_vars, full_command)
+        end
+
+        def execute_inline(command_string, env_vars = {})
```

Comment:
I know that a lot of tests are using inline commands for simplicity, but is there a real life case where we would ever want to test an inline command? Isn't this library for testing external scripts/functions?